front/src/applications/stdcm/components/StdcmForm/StdcmOpSchedule.tsx

```diff
@@ -1,83 +1,55 @@
-  onArrivalTypeChange: (arrivalType: ArrivalTimeTypes) => void;
-  onArrivalToleranceChange: ({
-    toleranceBefore,
-    toleranceAfter,
-  }: {
-    toleranceBefore: number;
-    toleranceAfter: number;
-  }) => void;
-  opScheduleTimeType: ArrivalTimeTypes;
-  opTimingData?: {
-    date: Date;
-    arrivalDate: string;
-    arrivalTime: string;
-    arrivalTimeHours: number;
-    arrivalTimeMinutes: number;
-  };
-  opToleranceValues: {
-    arrivalToleranceBefore: number;
-    arrivalToleranceAfter: number;
-  };
+  pathStep: Extract<StdcmPathStep, { isVia: false }>;
   opId: string;
   isOrigin?: boolean;
 };
 
-const defaultDate = (date?: Date) => {
-  const newDate = date ? new Date(date) : new Date();
-  newDate.setHours(0, 0, 0);
-  return newDate;
-};
-
-const StdcmOpSchedule = ({
-  disabled,
-  onArrivalChange,
-  onArrivalTypeChange,
-  onArrivalToleranceChange,
-  opTimingData,
-  opScheduleTimeType,
-  opToleranceValues,
-  opId,
-  isOrigin = false,
-}: StdcmOpScheduleProps) => {
+const StdcmOpSchedule = ({ disabled, pathStep, opId, isOrigin = false }: StdcmOpScheduleProps) => {
   const { t } = useTranslation('stdcm');
+  const dispatch = useAppDispatch();
+
+  const { updateStdcmPathStep } = useOsrdConfActions() as StdcmConfSliceActions;
   const { getSearchDatetimeWindow } = useOsrdConfSelectors();
   const searchDatetimeWindow = useSelector(getSearchDatetimeWindow);
 
-  const { arrivalDate, arrivalTime, arrivalTimeHours, arrivalTimeMinutes, arrivalToleranceValues } =
-    useMemo(() => {
-      const isArrivalDateValid =
-        opTimingData?.arrivalDate &&
-        isArrivalDateInSearchTimeWindow(opTimingData.date, searchDatetimeWindow);
-
+  const { arrivalTime, arrivalTimeHours, arrivalTimeMinutes } = useMemo(() => {
+    if (!pathStep.arrival) {
       return {
-        arrivalDate:
-          opTimingData && isArrivalDateValid
-            ? opTimingData.date
-            : defaultDate(searchDatetimeWindow?.begin),
-        arrivalTime: opTimingData?.arrivalTime,
-        arrivalTimeHours: opTimingData?.arrivalTimeHours,
-        arrivalTimeMinutes: opTimingData?.arrivalTimeMinutes,
-        arrivalToleranceValues: {
-          minusTolerance: opToleranceValues.arrivalToleranceBefore,
-          plusTolerance: opToleranceValues.arrivalToleranceAfter,
-        },
+        arrivalTime: undefined,
+        arrivalTimeHours: undefined,
+        arrivalTimeMinutes: undefined,
       };
-    }, [opTimingData, opToleranceValues, searchDatetimeWindow]);
+    }
+    return {
+      arrivalTime: dateToHHMMSS(pathStep.arrival, { withoutSeconds: true }),
+      arrivalTimeHours: pathStep.arrival.getHours(),
+      arrivalTimeMinutes: pathStep.arrival.getMinutes(),
+    };
+  }, [pathStep.arrival]);
+
+  const tolerances = useMemo(
+    () => ({
+      minusTolerance: pathStep.tolerances?.before || 0,
+      plusTolerance: pathStep.tolerances?.after || 0,
+    }),
+    [pathStep.tolerances]
+  );
```

Comment:
Seems like `tolerances` is always defined (we can also update `StdcmPathStep` type) and we can't set it to undefined.

```suggestion
  const tolerances = useMemo(
    () => ({
      minusTolerance: pathStep.tolerances.before,
      plusTolerance: pathStep.tolerances.after,
    }),
    [pathStep.tolerances]
  );
```